### PR TITLE
Unifiing messages, thread and issues API

### PR DESCRIPTION
**Pull request by PetrDlouhy**
I tried to unify conventions between Issues API and new Messages/Threads API. I made data type of `issues.order` parameter to `Symbol` with defined set of values. I also unified ordering fields to `order` and `order_direction` to preserve Issues compatibility. The `end_date`/`start_date` in issues can now accept `DateTime`.

### CODE REVIEW

**Comment by mvl22** [comment]
Thanks - very happy to consider improvements like this.

Just to say that there are currently API integrations currently at:

https://github.com/cyclestreets/cyclescape-issuemap/
https://github.com/cyclestreets/bikedata/

but I'm happy that these can be adjusted if necessary.

**Comment by PetrDlouhy** [comment]
@mvl22 This should not break current API integrations although you should test it before deployment.

File: app/api/route/issue_api.rb [hunk 1]
```diff
@@ -53,11 +54,11 @@ def post_or_get_issue
...
         case params[:order]
-        when 'vote_count'
+        when :vote_count
           scope = scope.plusminus_tally
-        when 'created_at', 'start_at'
-          scope = scope.order(params[:order] => :desc)
-        when 'size'
+        when :created_at, :start_at, :id
+          scope = scope.order(params[:order] => params[:order_direction])
+        when :size
```
Comment (nikolai-b) [inline app/api/route/issue_api.rb]: see https://github.com/cyclestreets/cyclescape/commit/b28f1d2753fe0fc5ab2c8f4bbaa1ce74d26eacbf you can now call 
```ruby
scope.order_by_size(params[:order_direction])
```

**Comment by PetrDlouhy** [comment]
@nikolai-b Fixed ordering direction by size, rebased.

**Comment by nikolai-b** [comment]
Merged (on command line) in https://github.com/cyclestreets/cyclescape/commit/0df33fdcb0c6760112bf912b5fa0ca40ac92e731

**Pull request closed without merging**